k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -756,9 +752,10 @@ Upload any local messages that are marked as PENDING_UPLOAD (Drafts, Sent, Trash
         String storeUri = account.getStoreUri();
         if (ImapStore.isStoreUriImap(storeUri)) {
             ImapSyncInteractor syncInteractor = new ImapSyncInteractor(account, folderName, listener, this);
-            syncInteractor.performSync(flagSyncHelper, messageDownloader, syncHelper);
+            syncInteractor.performSync(flagSyncHelper, messageDownloader, notificationController, syncHelper);
```

Comment:
how did you pick which of these parameters are passed to the constructor, and which to the method?

stuff like helper, downloader and controller seem more like constructor injected parameters, while folderName and listeners sound more suited as method parameters to me.